src/lib.rs

```diff
@@ -416,21 +416,37 @@ impl PurpleICQ {
     }
 
     fn group_chat_joined(&mut self, connection: &mut Connection, info: &ChatInfo) {
-        // Chat already joined.
         let mut account = connection.get_account();
-        let chat = purple::Chat::find(&mut account, &info.sn);
-        if chat.is_some() {
-            // TODO: Merge existing account info
-            return;
-        }
+        match purple::Chat::find(&mut account, &info.sn) {
+            Some(mut chat) => {
+                // The chat already exists.
+
+                // Should we replace the blist group?
+                if let Some(info_group) = &info.group {
+                    let should_replace_group = {
+                        match chat.get_group() {
+                            Some(mut chat_group) => !chat_group.get_name().eq(info_group),
+                            None => true,
+                        }
+                    };
+                    if should_replace_group {
+                        chat.add_to_blist(&mut self.get_or_create_group(Some(&info_group)), None);
```

Comment:
I guess this won't duplicate the existing chat and move it instead?

Reply:
Checked the code, should be good!

Reply:
Yep! I have read the implementation.